client/alloc_runner.go

```diff
@@ -425,6 +447,7 @@ OUTER:
 
 			// Check if we're in a terminal status
 			if update.TerminalStatus() {
+				taskDestroyEvent = structs.NewTaskEvent(structs.TaskTerminated)
```

Comment:
Don't set the `taskDestroyEvent` here as it will be set by the task runner (will do Killing and then Killed)